megamek/src/megamek/server/Server.java

```diff
@@ -1134,7 +1123,7 @@ public void sendServerChat(String message) {
     }
 
     void send(Packet packet) {
-        synchronized (connections) {
+        synchronized (serverLock) {
```

Comment:
Don't break MegaMek to fix a bug in MekHQ. Your changes to the synchronization introduce a new possibility of ConcurrentModificationExceptions.

Reply:
My apologies on my description. It's a MM bug that I found by testing with MHQ, not a bug in MHQ. It can be recreated by rapidly joining the same lobby with multiple MM setups.

Also, each ConMod at this point is a pre-existing issue that was being absorbed by our ancient connection setup. I will almost certainly cause new bugs with the connection changes (and will fix them), but these are old issues that the changes are raising.